R/layer-sf.R

```diff
@@ -35,6 +35,18 @@ LayerSf <- ggproto("LayerSf", Layer,
         self$mapping$geometry <- as.name(geometry_col)
       }
     }
+
+    # automatically determine the legend type
+    if (is.na(self$show.legend) || self$show.legend == TRUE) {
```

Comment:
We should not assume `show.legend` is a object of length 1. The conditions here should be wrapped `all()`, `any()` or `isTRUE()`.

Reply:
Yes. Writing code of the form `if (x == TRUE) {...}` is almost always bad in R, because `x` can be unexpectedly of length > 1 (even if nonsensical). Please get into the habit of writing `if (isTRUE(x)) {...}`.

Reply:
Got it~ Thanks for your kind instruction!